### PR TITLE
ci: add diff check action

### PR DESCRIPTION
**Pull request by burntcarrot**
Added a generate diff check for `go-service-azblob`.

Related: [#525](https://github.com/beyondstorage/go-storage/issues/525).

### CODE REVIEW

**Comment by Xuanwo** [comment]
Let‘s rename the file name to `diff-check.yml` to keep the same naming style.

File: .github/workflows/diff_check.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: "diff_check"
```
Comment (Xuanwo) [inline .github/workflows/diff_check.yml:1]: Use `Diff Check` here to keep the same naming style.

File: .github/workflows/diff_check.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+on: [ push,pull_request ]
+
+jobs:
+  diff_check:
+    name: diff_check
```
Comment (Xuanwo) [inline .github/workflows/diff_check.yml:7]: ditto

**Comment by burntcarrot** [comment]
Noted.

Just for a sanity check, can you inspect the `git diff` command? Is this what the expected behavior is supposed to be or do you need any modifications in the `git diff` command?

**Comment by Xuanwo** [comment]
> Just for a sanity check, can you inspect the `git diff` command? Is this what the expected behavior is supposed to be or do you need any modifications in the `git diff` command?

LGTM without any modifications

**Review by Xuanwo — APPROVED** [review]

**Comment by burntcarrot** [comment]
I'll move ahead with the same changes across all services! 😀

**Pull request merged**